tests/e2e/default/enketo/db-object-widget.wdio-spec.js

```diff
@@ -0,0 +1,82 @@
+const fs = require('fs');
+
+const utils = require('@utils');
+const userFactory = require('@factories/cht/users/users');
+const placeFactory = require('@factories/cht/contacts/place');
+const personFactory = require('@factories/cht/contacts/person');
+const commonPage = require('@page-objects/default/common/common.wdio.page');
+const loginPage = require('@page-objects/default/login/login.wdio.page');
+const genericForm = require('@page-objects/default/enketo/generic-form.wdio.page');
+const reportsPage = require('@page-objects/default/reports/reports.wdio.page');
+
+describe('DB Object Widget', () => {
+  const formId = 'db-object-widget';
+  const form = fs.readFileSync(`${__dirname}/forms/${formId}.xml`, 'utf8');
+  const formDocument = {
+    _id: `form:${formId}`,
+    internalId: formId,
+    title: `Form ${formId}`,
+    type: 'form',
+    context: { person: true, place: true },
+    _attachments: {
+      xml: {
+        content_type: 'application/octet-stream',
+        data: Buffer.from(form).toString('base64')
+      }
+    }
+  };
+
+  const places = placeFactory.generateHierarchy();
+  const districtHospital = places.get('district_hospital');
+  const area1 = places.get('health_center');
+  const area2 = placeFactory.place().build({
+    _id: 'area2',
+    name: 'area 2',
+    type: 'health_center',
+    parent: { _id: districtHospital._id }
+  });
+
+  const offlineUser = userFactory.build({ place: districtHospital._id, roles: [ 'chw' ] });
+  const personArea1 = personFactory.build({ parent: { _id: area1._id, parent: area1.parent } });
+  const personArea2 = personFactory.build({ name: 'Patricio', parent: { _id: area2._id, parent: area2.parent } });
+
```

Comment:
Please move this section out of the `describe` method. We have seen some really weird behaviors when we have code inside of the `describe` method that does not encapsulate a specific test and as part of our clean process of the test folders, we want to move all of those code sections outside of the `describe` method.

Reply:
I made this change but am not buying the idea yet, perhaps because I'm missing the context of `why` it's considered a better practice. 

> We have seen some really weird behaviors when we have code inside of the describe

What kind of weird behaviors? 🤔

> and as part of our clean process of the test folders

They are going to be closer to the "imports", I prefer having a clear separation between imports and test data. And we're going to lose the scoping of variables - like when having more than one `describe`.

Reply:
@lorerod could help us here because she read about that while she was doing an investigation related to a flaky test that we were having at some point.

Lore, could you please share more details about why it is not a good idea to add blocks of code inside of the `describe` method that doesn't belong to a specific `it`?

Reply:
Hi @tatilepizs @latin-panda! I came across this when trying to fix a flaky test. You can find the comment and mocha issues in slack [here](https://medic.slack.com/archives/C02H68VMD/p1699645534120339?thread_ts=1699568113.589189&cid=C02H68VMD) 
The unexpected behavior might be caused when using `async` in the `describe` block or having `async` methods called inside the `describe` block. The `describe` block is primarily used for grouping related test cases. If we need an `async` setup for the entire suite, consider using `before` and `beforeEach` and applying async to those functions.
Considering this, it might be a good practice, if you are calling an async function, to organize the code inside `before`, `beforeEach`, `after`, `afterEach`, and it, and use describe only for grouping purposes.

Reply:
I haven't thoroughly checked, but I believe there is no `async` code within this `describe` block. Therefore, it is recommended to leave it as it is. However, we should all exercise caution when adding code inside the `describe` block.

Reply:
Thanks for explaining! I get it, and it makes sense to me to make async code in `before`, `beforeEach`, `after`, `afterEach`, and `it`.
I have moved the test data declaration to inside the `describe` to make a clear separation with imports and make clear the scope.

Reply:
Thank you for the clarification @lorerod, I was confusing things 😅